integrations/sentry/main.py

```diff
@@ -9,34 +10,62 @@
 class ObjectKind(StrEnum):
     PROJECT = "project"
     ISSUE = "issue"
+    PROJECT_TAG = "project-tag"
+    ISSUE_TAG = "issue-tag"
 
 
-@ocean.on_resync(ObjectKind.PROJECT)
-async def on_resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    logic_settings = ocean.integration_config
+def init_client() -> SentryClient:
     sentry_client = SentryClient(
-        logic_settings["sentry_host"],
-        logic_settings["sentry_token"],
-        logic_settings["sentry_organization"],
+        ocean.integration_config["sentry_host"],
+        ocean.integration_config["sentry_token"],
+        ocean.integration_config["sentry_organization"],
     )
+    return sentry_client
+
+
+async def add_tags_to_issue(
+    sentry_client: SentryClient, issue: dict[str, Any]
+) -> dict[str, Any]:
+    tags = await sentry_client.get_issue_tags(issue["id"])
+    return {**issue, "__tags": tags}
+
+
+@ocean.on_resync(ObjectKind.PROJECT)
+async def on_resync_project(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    sentry_client = init_client()
 
     async for projects in sentry_client.get_paginated_projects():
-        logger.info(f"Received ${len(projects)} batch projects")
         yield projects
 
 
+@ocean.on_resync(ObjectKind.PROJECT_TAG)
+async def on_resync_project_tag(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    sentry_client = init_client()
+
+    async for projects in sentry_client.get_paginated_projects():
+        for project in projects:
+            tags = await sentry_client.get_project_tags(project["slug"])
+            project_tags = [{**project, "__tags": tag} for tag in tags]
+            yield project_tags
+
+
 @ocean.on_resync(ObjectKind.ISSUE)
-async def on_resync_issues(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    issues = []
-    logic_settings = ocean.integration_config
-    sentry_client = SentryClient(
-        logic_settings["sentry_host"],
-        logic_settings["sentry_token"],
-        logic_settings["sentry_organization"],
-    )
+async def on_resync_issue(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    sentry_client = init_client()
+
+    async for projects in sentry_client.get_paginated_projects():
+        for project in projects:
+            async for issues in sentry_client.get_paginated_issues(project["slug"]):
+                yield issues
+
+
+@ocean.on_resync(ObjectKind.ISSUE_TAG)
+async def on_resync_issue_tags(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    sentry_client = init_client()
 
     async for projects in sentry_client.get_paginated_projects():
```

Comment:
I think you can use the new cache decorator for projects, that way the integration won't need to refetch the projects for each kind. Check out azure devops integration for reference.